packages/untp-playground/src/contexts/TestReportContext.tsx

```diff
@@ -0,0 +1,112 @@
+
+interface TestReportContextType {
+  canGenerateReport: boolean;
+  canDownloadReport: boolean;
+  report: TestReport | null;
+  generateReport: (implementationName: string) => Promise<void>;
+  downloadReport: () => void;
+}
+
+const TestReportContext = createContext<TestReportContextType | undefined>(undefined);
+
+interface TestReportProviderProps {
+  children: React.ReactNode;
+  testResults: Partial<Record<CredentialType, TestStep[]>>;
+  credentials: Partial<Record<CredentialType, StoredCredential>>;
+}
+
+export function TestReportProvider({ children, testResults, credentials }: TestReportProviderProps) {
+  const [report, setReport] = useState<TestReport | null>(null);
+
+  const allowedStatuses = [TestCaseStatus.SUCCESS, TestCaseStatus.FAILURE, TestCaseStatus.WARNING];
+  const passStatuses = [TestCaseStatus.SUCCESS, TestCaseStatus.WARNING];
+
+  // Reset report when credentials change
+  useEffect(() => {
+    const credentialValues = Object.values(credentials);
+    const hasAnyCredential = credentialValues.some((cred) => cred && cred.decoded);
+    if (hasAnyCredential) {
+      setReport(null);
+    }
+  }, [credentials]);
+
+  // Allow report generation if there are any credentials with allowed statuses
+  const canGenerateReport =
+    testResults &&
+    Object.entries(testResults).every(([type, steps]) => {
+      if (!steps) return true;
+      const credential = credentials[type as CredentialType];
+
+      if (!credential || !credential.decoded) return true;
+
+      return steps.every((step) => allowedStatuses.includes(step.status));
+    }) &&
+    Object.entries(testResults).some(([type, steps]) => {
+      const credential = credentials[type as CredentialType];
+      return credential && credential.decoded && steps && steps.every((step) => allowedStatuses.includes(step.status));
+    });
+
+  const canDownloadReport = report !== null;
+
+  const handleGenerateReport = async (implementationName: string) => {
+    try {
+      const newReport = await generateReport({
+        implementationName,
+        credentials,
+        testResults,
+        passStatuses,
+      });
+
+      setReport(newReport);
+      toast.success('Report generated successfully');
+    } catch (error) {
+      console.error('Failed to generate report:', error);
+      toast.error(error instanceof Error ? error.message : 'Failed to generate report');
+    }
+  };
+
+  const downloadReport = () => {
+    if (!report) {
+      toast.error('No report available to download');
+      return;
+    }
+
+    try {
+      const filename = `untp-test-report-${report.implementation.name.toLowerCase().replace(/\s+/g, '-')}`;
+      downloadJson(report, filename);
+    } catch (error) {
+      console.error('Failed to download report:', error);
+      toast.error('Failed to download report');
+    }
+  };
+
+  return (
+    <TestReportContext.Provider
+      value={{
+        canGenerateReport,
+        canDownloadReport,
```

Comment:
It's still not clear to me why we don't just have the `Generate Report` button triggering the download.